### PR TITLE
Add select syscall handler and tests

### LINKED ISSUE #1721: Support select syscall
This syscall is commonly used in network applications.

### PR DESCRIPTION
**Pull request by robgjansen**
This PR provides support for the `select()` system call:
- adds new syscall handler in Shadow so native process calls to select no longer fail.
- re-enables legacy TCP test that can optionally use select as an event monitor.
- adds a new select-specific test to more completely test the select API against linux.

I decided to implement the handler in C first - it internally uses some code from our poll handler whose functionality is largely similar. I'll loop around to convert both the poll and select handlers to rust.

Closes #1721

### CODE REVIEW

**Review by stevenengler — APPROVED** [review]
Looks good to me! Now python's `sleep()` will work :)

File: src/test/select/test_select.rs [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+        if fd_is_set {
+            if fd_inval {
+                unsafe {
+                    // XXX: This code causes panics due to rust bounds checking on fd_set
+                    // (we are only allowed to set bits <= 1024)
+                    // TODO: I'm not sure the best way to test this case.
+                    libc::FD_SET(1025, &mut readfds);
+                    libc::FD_SET(1025, &mut writefds);
+                    libc::FD_SET(1025, &mut exceptfds);
+                }
```
[the remark this answers is not in the document]
Comment (robgjansen) [reply src/test/select/test_select.rs:265]: I was trying to test what linux would do if I pass a FD that is out of bounds of the fd_set. I think it just ignores those values as out of bounds as you say, so it won't really return an error or anything. So maybe not a useful thing to test after all?

[the remark this answers is not in the document]
Comment (stevenengler) [reply src/test/select/test_select.rs:265]: The `fd_set` is just a fixed sized buffer and `FD_SET` uses the fd as an offset in that buffer. If you use an fd greater than 1024, that offset will be outside of the bounds of the buffer and will cause undefined behaviour in the plugin (for example possibly a sigsegv).

https://github.com/bminor/glibc/blob/b92a49359f33a461db080a33940d73f47c756126/bits/select.h#L32-L33

```c
#define __FD_SET(d, s) \
  ((void) (__FDS_BITS (s)[__FD_ELT(d)] |= __FD_MASK(d)))
```

where:

```c
#define __NFDBITS	(8 * (int) sizeof (__fd_mask))
#define	__FD_ELT(d)	((d) / __NFDBITS)
#define	__FD_MASK(d)	((__fd_mask) (1UL << ((d) % __NFDBITS)))
```

Also I think this 1024 limit is a glibc limit and not a linux limit:

> The Linux kernel imposes no fixed limit, but the glibc implementation makes fd_set a fixed-size type,  with  FD_SETSIZE defined as 1024, and the FD_*() macros operating according to that limit.

If you wanted to pass fds greater than 1024, I think you'd need to create your own buffer and set the bits yourself, and not use the glibc macros.

[the remark this answers is not in the document]
Comment (robgjansen) [reply src/test/select/test_select.rs:265]: Yeah. I don't think I should test a buffer overrun here. It's kinda out of scope.

[the remark this answers is not in the document]
Comment (robgjansen) [reply src/test/select/test_select.rs:265]: I added a test for EBADF without exceeding the `fd_set` memory boundaries.

**Pull request merged**